lib/formulator.ex

```diff
@@ -2,31 +2,82 @@ defmodule Formulator do
   use Phoenix.HTML
   alias Formulator.HtmlError
 
+  @doc  """
+  Returns an html input with an associated label.
+  If there are errors associated with the field, it will also output a span
+  tag with the errors.
+
+  ## Options
+
+    * `:label` - when false, does not create a label tag. Instead, an `aria-label`
+    attribute is added to the input to improve accessibility.
+    * `:label_text` - when present, the given string is used as the label text.
+
+  ## Examples
+
+  Basic input:
+      <%= input form, :email %>
+      #=> <label for="user_email">Email</label>
+      #=> <input id="user_name" name="user[email]" type="text" value="">
+
+  Without a label:
+      <%= input form, :email, label: false %>
+      #=> <input id="user_name" name="user[email]" aria-label="email" type="text" value="">
+  """
+
+  @spec input(Phoenix.HTML.Form.t, atom, []) :: binary
   def input(form, field, options \\ []) do
+    cond do
+      {:label, false} in options -> input_without_label(form, field, options)
+      true -> input_with_label(form, field, options)
+    end
+  end
+
+  defp input_without_label(form, field, options) do
+    options = options ++ build_aria_label(field, options[:label_text])
+    error = html_error(form, field)
+    [
+      build_input(form, field, options, error)
+    ] ++ error.html
+  end
+
+  defp input_with_label(form, field, options) do
     error = html_error(form, field)
     build_html(form, field, options, error)
   end
 
+  defp build_aria_label(field, label_text) do
+    ["aria-label": label_text || field |> to_string |> String.replace("_", " ")]
```

Comment:
Could the `field |> to_string |> String.replace("_", " ")` part be extracted out to be a little clearer?